coldfront/core/allocation/views.py

```diff
@@ -414,6 +416,136 @@ def get_context_data(self, **kwargs):
 
         return context
 
+class AllocationListItem:
+    id: int
+    project_id: int
+    project_name: str
+    resource_name: str
+    department_number: str
+    allocation_status: str
+    pi_last_name: str
+    pi_first_name: str
+    pi_user_name: str
+    itsd_ticket: str
+    file_path: str
+    service_rate: str
+
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+
+class AllocationTableView(LoginRequiredMixin, ListView):
+
+    model = Allocation
+    template_name = 'allocation/allocation_table_view.html'
+    context_object_name = 'allocation_list'
+    paginate_by = 25
+
+    def get_queryset(self):
+
+        view_list: List[AllocationListItem] = []
+
+        allocation_search_form = AllocationSearchForm(self.request.GET)
+        
+
+        if allocation_search_form.is_valid():
```

Comment:
Only a suggestion, feel free to disregard if you don't care for it:
```suggestion
        if !allocation_search_form.is_valid():
            return view_list
```

Primary reason is just to reduce indentation hell.